src/_zkapauthorizer/tests/test_storage_protocol.py

```diff
@@ -180,6 +184,16 @@ def is_successful_write():
     )
 
 
+def synchronize_clocks(from_clock, to_clock):
+    """
+    Make the time on ``to_clock`` match the time on ``from_clock``.
+
+    :param Clock from_clock: The clock from which to take the time.
+    :param Clock to_clock: The clock to which to write a time.
+    """
+    to_clock.advance(from_clock.seconds() - to_clock.seconds())
```

Comment:
That is very clever. :)  Also I had not previously understood what `runner_strategy` would give you in our usage.  I don't think of the `TestCase` instance as a "runner" so I think that threw me off.

I did have a thought for another approach... Create a `DelegatedClock` that wraps another clock and delegates all its calls there and also supports mutating the delegatee (then just swap in the Hypothesis-supplied `Clock` at the start of each test).  Since all of this `Clock`-based code is probably here for a while, perhaps that's worth pursuing.  I'll try to switch over to that idea and see how well or poorly it goes.

Reply:
Okay that `DelegatedClock` thing was silly.  Instead we can just throw out the extra clock and have Hypothesis generate a posix timestamp, then advance `self.clock` to that.  Then, since there's only one clock, there's no synchronization to do afterwards.